feat: a toString() method for enumerated values in Java

subhajit20 opened the pull request:
<!--- Provide a formatted commit message describing this PR in the Title above -->
<!--- See our DEVELOPERS guide below: -->
<!--- https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format -->
# Closes https://github.com/accordproject/concerto/issues/100
<!--- Provide an overall summary of the pull request -->

### Changes
<!--- More detailed and granular description of changes -->
<!--- These should likely be gathered from commit message summaries -->
- <ONE> override the toString method inside the javavisitor enumDeclaration function
- <TWO> Updated snapshots for required changes
- <THREE> Added test for the changes


### Author Checklist
- [X] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [ ] Vital features and changes captured in unit and/or integration tests
- [ ] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `main` from `fork:branchname`

subhajit20 committed:
Added toString method to enumdeclaration in javavisior

Signed-off-by: Subhajit Ghosh <subhajitstd07@gmail.com>

subhajit20 committed:
Added minor changes

Signed-off-by: Subhajit Ghosh <subhajitstd07@gmail.com>

subhajit20 committed:
Added some space to the return statement

Signed-off-by: Subhajit Ghosh <subhajitstd07@gmail.com>

subhajit20 commented:
hey @mttrbrts @sanketshevkar can you please review and give some feedback

subhajit20 committed:
Added test

Signed-off-by: Subhajit Ghosh <subhajitstd07@gmail.com>

sanketshevkar commented on test/codegen/__snapshots__/codegen.js.snap:
```diff
@@ -1196,6 +1208,10 @@ import com.fasterxml.jackson.annotation.*;
 public enum LaptopMake {
    Apple,
    Microsoft,
+   @Override
```
I am not sure if enums in java store functions.

subhajit20 replied:
Yes it does, because I have created one java program where I declared a enum and inside it defined a function and it worked totally final without any

sanketshevkar replied:
Okay, thanks for clarification. I'm not much proficient in Java. @mttrbrts can you please review this?

subhajit20 replied:
```
public class Main{
    public enum State {
        MA,
        NY,
        CO,
    
        // Override the toString() method for the entire enum type
        @Override
        public String toString() {
            return name();
        }
    }
    public static void main(String[] args) {
        // Using the enum constants
        State state1 = State.MA;
        State state2 = State.NY;
        String ma = state1.toString();
        
        // Printing the enum constants
        System.out.println("State 1: " + ma); // This will call the overridden toString() method
    }
}

```

subhajit20 replied:
Hey @sanketshevkar Have one question, How can I pass extra parameters like` parameters.filewriter` as show in the code.. I mean to say can we pass other parameter like this `parameters.x` ?
https://github.com/accordproject/concerto-codegen/blob/7c14d706387e371d22c36ccdb17fda08b4a2f1e7/lib/codegen/fromcto/java/javavisitor.js#L142-L146

subhajit20 replied:
> yes, you can. it is an options object on the visitor. (answered by @dselman on the Tech WG call)

Okay ... I am assuming is that something like this `parameters.options`

subhajit20 replied:
Hi! is there any update or any feedback on this pr? @dselman @mttrbrts

mttrbrts committed:
chore: merge main into feature branch, update snapshots

Merges upstream main into feature/javavisitor_enumDeclaration_toString
to resolve conflicts. Snapshot file was regenerated using current main's
test naming convention ('versioned CTO') with the new toString() output
included.

Signed-off-by: Matt Roberts <mttrbrts@users.noreply.github.com>

mttrbrts committed:
fix: add semicolon after last enum constant before toString() method

Java requires a semicolon terminating the enum constant list before any
method bodies. Without it, javac errors with '<identifier> expected' on
the @Override annotation.

Signed-off-by: Matt Roberts <mttrbrts@users.noreply.github.com>

mttrbrts committed:
fix: handle empty enum constants in Java toString() generation

Empty enums (zero constants) also need a semicolon before method
declarations per the Java Language Specification (JLS 8.9.1).

Signed-off-by: Matt Roberts <mttrbrts@users.noreply.github.com>
